src/docs/markdown/caddyfile/common-uses.md

```diff
@@ -0,0 +1,127 @@
+---
+title: Common Caddyfile Uses
+---
+
+# Common Caddyfile Uses
+
+This page presents some complete, minimal Caddyfile configurations for common use cases, which you might find helpful when building your own configs.
+
+These are not drop-in solutions; you will have to customize your domain name, ports/sockets, directory paths, etc. They are intended to illustrate some of the most common configuration patterns.
+
+#### Menu
+
+- [Static file server](#static-file-server)
+- [Reverse proxy](#reverse-proxy)
+- [PHP](#php)
+- [Redirect `www.` subdomain](#redirect-www-subdomain)
+- [Trailing slashes](#trailing-slashes)
+
+
+## Static file server
+
+```caddy
+example.com
+
+root * /var/www
+file_server
+```
+
+As usual, the first line is the site address. The [`root` directive](/docs/caddyfile/directives/root) specifies the path to the root of the site (the `*` means to match all requests, so as to disambiguate from a [path matcher](/docs/caddyfile/matchers#path-matchers))&mdash;change the path to your site if it isn't the current working directory. Finally, we enable the [static file server](/docs/caddyfile/directives/file_server).
+
+
+## Reverse proxy
+
+Proxy all requests:
+
+```caddy
+example.com
+
+reverse_proxy localhost:5000
+```
+
+Only proxy requests having a path starting with `/api/` and serve static files for everything else:
+
+```caddy
+example.com
+
+root * /var/www
+reverse_proxy /api/* localhost:5000
+file_server
+```
+
+
+## PHP
+
+With a PHP FastCGI service running, something like this works for most modern PHP apps:
+
+```caddy
+example.com
+
+root * /var/www
+php_fastcgi /blog/* localhost:9000
+file_server
+```
+
+Customize the site root and path matcher accordingly; this example assumes PHP is only in the `/blog/` subdirectory&mdash;all other requests will be served as static files.
+
+The [`php_fastcgi` directive](/docs/caddyfile/directives/php_fastcgi) is actually just a shortcut for [several pieces of configuration](/docs/caddyfile/directives/php_fastcgi#expanded-form).
+
+
+## Redirect `www.` subdomain
+
+To **add** the `www.` subdomain with an HTTP redirect:
+
+```caddy
+example.com {
+	redir https://www.example.com{uri}
```

Comment:
Do we want to consider making it like this?

```suggestion
	redir https://www.{host}{uri}
```

Reply:
I'd rather keep the examples as simple as possible. Also, this would omit the port (if non-standard).